### ISSUE TITLE
Support for creation timeout in aws_elasticsearch_domain

### ISSUE THREAD

**Issue by mnowrot** [hashicorp/terraform-provider-aws#15707]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

aws_elasticsearch_domain resource should support creation timeout (i.e. it should fail terraform apply, if the ES domain is not up within the given timeout)

### New or Affected Resource(s)

* aws_elasticsearch_domain

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.

resource "aws_elasticsearch_domain" "mydomain" {
    timeouts {
        create = "30m"
    }
}
```

### References

* No known references

**Comment by ellisroll-b** [comment]
AWS is having issues, and taking longer than 1 hour... so TF limitation is breaking our deploys...

**Comment by breathingdust** [comment]
This was closed via #19713
